### DISCUSSION TITLE
Predefining arguments through closure as an alternative to .bind method?

### DISCUSSION THREAD

**Discussion by sasatatar** [next-safe-action/next-safe-action#158] [General]
Let's say I have a simple server action that takes `postId` and `message` as payload:

```ts
// send-message.actions.ts file
"use server";

import { authAction } from "@/lib/safe-action";

const payloadSchema = z.object({
  postId: z.string(),
  message: z.string(),
});

export const addCommentAction = authAction(payloadSchema, (payload: { postId: string, message: string }) => {
    // some logic to add a comment...
});
```

And let's say I have a simple form component that collects the message input prop that gets called on form submit event with the  `message` entered by the user, I would need to, as suggested by the docs, split up the parameters and use `.bind` to pass the `postId` first, and then use the bound function within the component to pass the message.

This however is not going to work if we want to use `next-safe-action` as it only works with a single parameter passed to the action.
To solve this problem, I came up with this workaround:

```ts
// parent server component with access to `postId`
({ postId }: { postId: string }) => {

  const addComment = (message: string) => {
    "use server";
    // postId is available through closure:
    return addCommentAction({ postId, message })
  };
  
  return <AddCommentForm onSubmit={addComment} />
}
```

I wanted to see what you all think about this approuch, vs the `.bind` method? Are there some drawbacks to this, or am I perhaps missing something, as to why we should not be doing it this way?

Many thanks in advance!

**Comment by TheEdoRan** [comment]
v7 supports bind arguments, see related docs [here](https://next.next-safe-action.dev/docs/safe-action-client/bind-arguments).

**Reply by sasatatar** [reply]
I forgot to mention, I am aware that v7 supports this (btw. the new version looks amazing, looking forward to it!), but I started migrating our actions before we upgraded to Next.js 15, and most likely, we will not migrate to Next.js 15 until it's stable, so I had to find a solution to make it work with current version of next-safe-action (v6). Also, I am wondering if this is actually a viable alternative to binding arguments in future as well?
